### PR TITLE
Added some stats during pull request

### PR DESCRIPTION
**Pull request by andjelx**
Added some verbosity to pull request (which classes are added, removed, updated)

### CODE REVIEW

File: syncano_cli/sync/project.py [hunk 1]
```diff
@@ -48,13 +48,34 @@ def update_from_instance(self, instance, all=False, classes=None,
...
         LOG.info("Pulling instance data from syncano")
+        classesI = self.classes
         classes = classes or self.classes.keys()
         scripts = scripts or set(s['label'] for s in self.scripts)
         if all:
             classes = None
             scripts = None
         self.classes = pull_classes(instance, classes)
         self.scripts = pull_scripts(instance, scripts)
+        def cmpDicts(d1, d2):
```
Comment (forgems) [inline syncano_cli/sync/project.py]: cmpDicts shouldn't be an inner function.
Also we don't use camelCase in this project.

File: syncano_cli/sync/project.py [hunk 2]
```diff
@@ -48,13 +48,34 @@ def update_from_instance(self, instance, all=False, classes=None,
                              scripts=None):
         """Updates project data from instances"""
         LOG.info("Pulling instance data from syncano")
+        classesI = self.classes
```
Comment (forgems) [inline syncano_cli/sync/project.py]: classesI doesn't tell anything about this variable. Use prev_classes or saved_classess.

File: syncano_cli/sync/project.py [hunk 3]
```diff
@@ -48,13 +48,34 @@ def update_from_instance(self, instance, all=False, classes=None,
...
+            intersect_keys = d1_keys.intersection(d2_keys)
+            added = d1_keys - d2_keys
+            removed = d2_keys - d1_keys
+            modified = {o : (d1[o], d2[o]) for o in intersect_keys if d1[o] != d2[o]}
+            same = set(o for o in intersect_keys if d1[o] == d2[o])
+            return same, added, removed, modified
+        state = ("Not changed", "Added", "Removed", "Updated" )
+        if self.classes:
+            LOG.info("Stats for classes")
+            for i,s in enumerate(cmpDicts(self.classes, classesI)):
```
Comment (forgems) [inline syncano_cli/sync/project.py]:
```
for info, classes in zip(state, cmpDicts(self.classes, classesI)):
    if classes:
        LOG.info('%s: %s', info, ','.join(classes))
```

File: syncano_cli/sync/project.py [hunk 4]
```diff
@@ -48,13 +48,35 @@ def update_from_instance(self, instance, all=False, classes=None,
...
+        prev_classes = self.classes
         classes = classes or self.classes.keys()
         scripts = scripts or set(s['label'] for s in self.scripts)
         if all:
             classes = None
             scripts = None
         self.classes = pull_classes(instance, classes)
         self.scripts = pull_scripts(instance, scripts)
+
+        def cmpDicts(d1, d2):
```
Comment (forgems) [inline syncano_cli/sync/project.py]: d1 should be named old and d2 should be new.

File: syncano_cli/sync/project.py [hunk 5]
```diff
@@ -48,13 +48,35 @@ def update_from_instance(self, instance, all=False, classes=None,
...
+            """
+            Compare two dicts returning added, removed, modified, same
+            """
+            d1_keys = set(d1.keys())
```
Comment (forgems) [inline syncano_cli/sync/project.py]: d1_keys, d2_keys = set(d1), set(d2 or ())

File: syncano_cli/sync/project.py [hunk 6]
```diff
@@ -48,13 +48,35 @@ def update_from_instance(self, instance, all=False, classes=None,
...
+            if not d2:
+                return None, d1, None, None
+            d2_keys = set(d2.keys())
+            intersect_keys = d1_keys.intersection(d2_keys)
+            added = d1_keys - d2_keys
+            removed = d2_keys - d1_keys
+            modified = {o: (d1[o], d2[o]) for o in intersect_keys if d1[o] != d2[o]}
+            same = set(o for o in intersect_keys if d1[o] == d2[o])
+            return same, added, removed, modified
```
Comment (forgems) [inline syncano_cli/sync/project.py]: Be consistent in what you return. You return 3 sets and a dictionary.

File: syncano_cli/sync/project.py [hunk 7]
```diff
@@ -48,13 +48,35 @@ def update_from_instance(self, instance, all=False, classes=None,
...
         self.classes = pull_classes(instance, classes)
         self.scripts = pull_scripts(instance, scripts)
+
+        def cmpDicts(d1, d2):
+            """
+            Compare two dicts returning added, removed, modified, same
+            """
+            d1_keys = set(d1.keys())
+            if not d2:
+                return None, d1, None, None
```
Comment (forgems) [inline syncano_cli/sync/project.py]: In other place you return 3 sets and iterable, and here you return None, d1, None, None. Be consistent in returning objects with the same type. So If you really insist on this `if` it should be `return set(), set(), set(), set()`

File: syncano_cli/sync/project.py (hunk 4 above)
Comment (forgems) [inline syncano_cli/sync/project.py]:
```
def diff_dicts(old, new):
    old_keys = set(old or ())
    new_keys = set(new or ())
    common = new_keys & old_keys
    added = new_keys - old_keys
    removed = old_keys - new_keys
    same = set(k for k in common if old[k] == new[k])
    return same, added, removed, common - same
```

File: syncano_cli/sync/utils.py [hunk 8]
```diff
@@ -0,0 +1,18 @@
+# coding=UTF8
+from __future__ import print_function, unicode_literals
+
+
+def compare_dicts(new, old):
```
Comment (forgems) [inline syncano_cli/sync/utils.py:5]: I gave you whole code for this function in PR comment. Why haven't you used it ?
So again

```
def diff_dicts(old, new):
    old_keys = set(old or ())
    new_keys = set(new or ())
    common = new_keys & old_keys
    added = new_keys - old_keys
    removed = old_keys - new_keys
    same = set(k for k in common if old[k] == new[k])
    return same, added, removed, common - same
```

Comment (andjelx) [inline syncano_cli/sync/utils.py:5]: Fixed

Comment (andjelx) [inline syncano_cli/sync/utils.py]: Fixed.

**Comment by forgems** [comment]
Nice job ;) @opalczynski, could you take a look at it and merge it ?:)

**Pull request merged**